### PR TITLE
docker rootless info

### PR DESCRIPTION
**Pull request by arronlacey**
updated to reflect that docker can now be run rootless

### CODE REVIEW

File: book/website/reproducible-research/renv/renv-containers.md [hunk 1]
```diff
@@ -54,7 +54,7 @@ That person can then use the image to generate a container containing a working
 
 There are many tools available for creating and working with containers.
 We will focus on Docker, which is widely used, but be aware that others such as Singularity also exist.
-Singularity is sometimes preferred for use on high-performance computing systems as it does not need `sudo` permissions to be run, while Docker does.
+Singularity is sometimes preferred for use on high-performance computing systems as it does not need `sudo` permissions to be run, while up until April 2020 Docker did (please see the [Docker without root access](#docker-without-root-access) section)
```
Comment (RaoOfPhysics) [inline book/website/reproducible-research/renv/renv-containers.md:57]:
```suggestion
Singularity is sometimes preferred for use on high-performance computing systems as it does not need `sudo` permissions to be run, while up until April 2020 Docker did (please see the {ref}`rr-renv-containers-rootless` section).
```

File: book/website/reproducible-research/renv/renv-containers.md [hunk 2]
```diff
@@ -395,22 +395,39 @@ Once the researcher is done, they can close the container as normal.
...
+ 
+ The underyling difference between Docker and rootless mode is that perviously any system running docker had a daemon running as `uid0` that creates and owns all images, but with rootless mode the user creates and owns any images that they initialize.
+ To install and run the rootless version of Docker as a non-root user, use the following commands:
+
+```
+dockerd-rootless-setuptool.sh install
+docker run -d --name dind-rootless --privileged docker:20.10-dind-rootless
+```
+
+The following prequisites are required to run Docker rootless: `newuidmap` and `newgidmap`.
```
Comment (RaoOfPhysics) [inline book/website/reproducible-research/renv/renv-containers.md:421]:
```suggestion
The following prequisites, which are part of the [`shadow-utils`](https://github.com/shadow-maint/shadow) package are required to run Docker rootless: `newuidmap` and `newgidmap`. 
```

File: book/website/reproducible-research/renv/renv-containers.md [hunk 3]
```diff
@@ -395,22 +395,39 @@ Once the researcher is done, they can close the container as normal.
...
 If, when deleting a container, a `-v` is included after `rm` in `sudo docker rm container_ID`, any volumes associated with the container will also be deleted.
 
+(rr-renv-containers-rootless)=
+## Docker without root access
+
+Up until April 2020, the only way to run Docker was with root access. 
+"Rootless" mode was made available as part of the [v20.10](https://docs.docker.com/engine/security/rootless/) release.
+ Rootless mode is currently only avaliable on Linux and requires an initial install of Docker >= v20.10. 
+ 
+ The underyling difference between Docker and rootless mode is that perviously any system running docker had a daemon running as `uid0` that creates and owns all images, but with rootless mode the user creates and owns any images that they initialize.
```
Comment (RaoOfPhysics) [inline book/website/reproducible-research/renv/renv-containers.md:413]:
```suggestion
 The underyling difference between Docker without and with rootless mode is that perviously any system running Docker had a daemon running as `uid0` that creates and owns all images, but with rootless mode the user creates and owns any images that they initialize.
```

File: book/website/reproducible-research/renv/renv-containers.md [hunk 4]
```diff
@@ -395,22 +395,39 @@ Once the researcher is done, they can close the container as normal.
...
+ To install and run the rootless version of Docker as a non-root user, use the following commands:
+
+```
+dockerd-rootless-setuptool.sh install
+docker run -d --name dind-rootless --privileged docker:20.10-dind-rootless
```
Comment (RaoOfPhysics) [inline book/website/reproducible-research/renv/renv-containers.md:418]: Does `20.10` refer to the installed version of Docker?

**Review by RaoOfPhysics — APPROVED** [review]
Hey, @arronlacey. Made some changes for you to consider, and also have a question about the Docker command for `rootless`. I’ve provisionally approved the PR, so you can go ahead and hit merge if you’re happy with the changes or need to make any further ones.

File: book/website/reproducible-research/renv/renv-containers.md (hunk 1 above)
Comment (arronlacey) [reply book/website/reproducible-research/renv/renv-containers.md:57]: 👍

File: book/website/reproducible-research/renv/renv-containers.md (hunk 2 above)
Comment (arronlacey) [reply book/website/reproducible-research/renv/renv-containers.md:421]: 👍

File: book/website/reproducible-research/renv/renv-containers.md (hunk 3 above)
Comment (arronlacey) [reply book/website/reproducible-research/renv/renv-containers.md:413]: 👍

File: book/website/reproducible-research/renv/renv-containers.md (hunk 4 above)
Comment (arronlacey) [reply book/website/reproducible-research/renv/renv-containers.md:418]: that's correct!

**Comment by arronlacey** [comment]
> Hey, @arronlacey. Made some changes for you to consider, and also have a question about the Docker command for `rootless`. I’ve provisionally approved the PR, so you can go ahead and hit merge if you’re happy with the changes or need to make any further ones.

Thanks so much for this @RaoOfPhysics - your edits all look good to me.

**Comment by RaoOfPhysics** [comment]
Hi @arronlacey! Sorry I didn’t look at this earlier. Looks like the suggestions didn’t make it to the final version. I’m not familiar with how they work but I guess you have to accept them and add a new commit? I can fix this in a separate PR, though. :)

**Pull request merged**